src/ocl/rnnocl.cpp

```diff
@@ -87,6 +87,20 @@ void checkGemmStatusAndLog(miopenStatus_t gemm_status)
     }
 }
 
+GemmBackend_t GetGemmBackend(const miopen::Handle& handle, miopenDataType_t dataType)
+{
+#if MIOPEN_USE_HIPBLASLT
+    // Only use the hipblaslt backend when device is MI300, and the datatype is half.
+    // Otherwise, default to rocblas.
+    // Note: environment variable can be used to force a specific backend.
+    return handle.GetDeviceName() == "gfx942" && dataType == miopenDataType_t::miopenHalf
```

Comment:
I forbid making this change. Only optional by ENV switch, default is rocblas.
I don't see any significant improvements in the local tests.
Moreover, in multithreaded mode current develop shows better performance.
Develop 19.802ms , this PR: 22.89ms. the new value is 15% slower
./MIOpenDriver rnnfp16 -n 1024 -W 224 -H 1000 -l 8 -b 1 -m lstm -p 0 -r 0 -k 32 -F 1 -c 0 -t 1 -w 1 -V 0 --iter 10

Reply:
Curious why we are getting different results.
When we ran our tests there was a minor improvement using hipBLASLt over rocBLAS 3-4% it seemed. 
I would be fine with leaving it only accessible behind the environment variable for now though.

Reply:
@shurale-nkn oh, I think it could possibly be due to the large performance hit the first time you use hipBLASLt in an application. Finding the heuristic has a long load time the first time running it, but the second run is much quicker, and that first load is ignored in the benchmarks we were running.

Reply:
Re-ran the above to check for me, and I get the following with default rocBLAS & hipBLASLt installed:

**Branch hipBLASLt:**
MIOPEN_GEMM_ENFORCE_BACKEND=5 ./bin/MIOpenDriver rnnfp16 -n 1024 -W 224 -H 1000 -l 8 -b 1 -m lstm -p 0 -r 0 -k 32 -F 1 -c 0 -t 1 -w 1 -V 0 --iter 10
MIOpenDriver rnnfp16 -n 1024 -W 224 -H 1000 -l 8 -b 1 -m lstm -p 0 -r 0 -k 32 -F 1 -c 0 -t 1 -w 1 -V 0 --iter 10
length of data sequence == 1 is short than time sequence == 32, padding the rest of data sequence with 1024
length of data sequence == 1 is short than time sequence == 32, padding the rest of data sequence with 1024
PRNG seed: 12345678
GPU Kernel Time Forward RNN Elapsed: 25.389431 ms
Wall-clock Time Forward RNN Elapsed: 25.442097 ms

**Branch rocBLAS:**
MIOPEN_GEMM_ENFORCE_BACKEND=1 ./bin/MIOpenDriver rnnfp16 -n 1024 -W 224 -H 1000 -l 
8 -b 1 -m lstm -p 0 -r 0 -k 32 -F 1 -c 0 -t 1 -w 1 -V 0 --iter 10
MIOpenDriver rnnfp16 -n 1024 -W 224 -H 1000 -l 8 -b 1 -m lstm -p 0 -r 0 -k 32 -F 1 -c 0 -t 1 -w 1 -V 0 --iter 10
length of data sequence == 1 is short than time sequence == 32, padding the rest of data sequence with 1024
length of data sequence == 1 is short than time sequence == 32, padding the rest of data sequence with 1024
PRNG seed: 12345678
GPU Kernel Time Forward RNN Elapsed: 31.276720 ms
Wall-clock Time Forward RNN Elapsed: 31.345469 ms

**Develop rocBLAS**:
MIOPEN_GEMM_ENFORCE_BACKEND=1 ./bin/MIOpenDriver rnnfp16 -n 1024 -W 224 -H 1000 -l 8 -b 1 -m lstm -p 0 -r 0 -k 32 -F 1 -c 0 -t 1 -w 1 -V 0 --iter 10
MIOpenDriver rnnfp16 -n 1024 -W 224 -H 1000 -l 8 -b 1 -m lstm -p 0 -r 0 -k 32 -F 1 -c 0 -t 1 -w 1 -V 0 --iter 10
length of data sequence == 1 is short than time sequence == 32, padding the rest of data sequence with 1024
length of data sequence == 1 is short than time sequence == 32, padding the rest of data sequence with 1024
PRNG seed: 12345678
GPU Kernel Time Forward RNN Elapsed: 29.586761 ms
Wall-clock Time Forward RNN Elapsed: 29.644012 ms

This is with the following versions:
- hipBLASLt 0.7.0 Tweak 1549b021
- rocBLAS 4.1.0 Tweak cefa4a9b-dirty

Reply:
I'm using tuned rocblas from develop
![image](https://github.com/ROCm/MIOpen/assets/37591772/ed33b3b7-4a92-427a-b79e-7b0216ebff4d)

Reply:
Ah that explains it.
Ill double check with latest hipBLASLt, but I will swap to make it only usable through the environment variables.

Reply:
Well wait, when I tested it was with a tuned version of rocBLAS:

#define ROCBLAS_VERSION_MAJOR       4
#define ROCBLAS_VERSION_MINOR       3
#define ROCBLAS_VERSION_PATCH       0
#define ROCBLAS_VERSION_TWEAK       18feb608
#define ROCBLAS_TENSILE_COMMIT_ID   "18feb6085f70551903db177797a90d93b88e688b" , "1e5f2423ddf5594077e883da138f7efaf84f77bd"

Let me check again with this and latest hipBLASLt.

Reply:
Even with the newer version of rocBLAS I am still getting similar results:

MIOPEN_GEMM_ENFORCE_BACKEND=1 ./bin/MIOpenDriver rnnfp16 -n 1024 -W 224 -H 1000 -l 8 -b 1 -m lstm -p 0 -r 0 -k 32 -F 1 -c 0 -t 1 -w 1 -V 0 -iter 10
MIOpenDriver rnnfp16 -n 1024 -W 224 -H 1000 -l 8 -b 1 -m lstm -p 0 -r 0 -k 32 -F 1 -c 0 -t 1 -w 1 -V 0 -iter 10
length of data sequence == 1 is short than time sequence == 32, padding the rest of data sequence with 1024
length of data sequence == 1 is short than time sequence == 32, padding the rest of data sequence with 1024
PRNG seed: 12345678
GPU Kernel Time Forward RNN Elapsed: 23.978241 ms
Wall-clock Time Forward RNN Elapsed: 24.052076 ms

They seem to be about the same to me though, and I will put it back to using rocBLAS as the default.

Reply:
Alright @shurale-nkn, I swapped it so now it should only be usable via the environment variable for now.

Are you good with these changes?